File: Python_LeetCode_Solutions/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        num_dict = {}
        for i, num in enumerate(nums):
            diff = target - num
            # Check if the difference is already in the dictionary
            if diff in num_dict:
                return [num_dict[diff], i]
            # Store the number with its index
            num_dict[num] = i
        return []

File: Java_LeetCode_Solutions/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i,j;
        int[] ans= new int[2];
        int start=o;
        int end= nums.length-1;

        while(start<end){
            int sum= nums[start]+nums[end];
            if(sum==target){
                ans[0]=start;
                ans[1]=end;
            }
            else if(sum>target){
                end--;
            }
            else{
                start++;
            }
        }
        
    }
    return ans;
}
